cargo-shuttle/src/cargo_builder.rs

```diff
@@ -0,0 +1,298 @@
+use anyhow::Result;
+use crates_index::Index;
+use std::collections::BTreeMap;
+use toml_edit::{value, Document, Value};
+
+#[allow(dead_code)]
+pub enum CargoSection {
+    Dependency(Dependency),
+    Package,
+}
+
+#[derive(Debug, Default)]
+pub struct CargoBuilder {
+    packages: BTreeMap<String, Value>,
+    dependencies: BTreeMap<String, BTreeMap<String, Value>>,
+}
+
+#[derive(Clone)]
+pub struct Dependency {
+    name: String,
+    version: Option<String>,
+}
+
+impl Dependency {
+    pub fn new(name: String, version: Option<String>) -> Self {
+        Dependency { name, version }
+    }
+
+    pub fn get_latest_version(&self) -> String {
+        match &self.version {
+            Some(x) => x.to_string(),
+            None => {
+                let index = Index::new_cargo_default().unwrap();
+                let crate_ver = index
+                    .crate_(&self.name)
+                    .unwrap_or_else(|| panic!("Could not find package {} in registry", self.name));
+
+                crate_ver
+                    .highest_normal_version()
+                    .unwrap()
+                    .version()
+                    .to_string()
```

Comment:
Are we still covering the case where we want to fetch a pre-release version? For example for rocket we want init to start with rocket 0.5rc-3, not the latest stable release which is 0.4.x.